### PR TITLE
Upgrade docs from 2.4.1 to 2.4.2

### PR DESCRIPTION
**Pull request by gonzalo-bulnes**
## Status
Ready for review; in draft mode until release object is live

Related to https://github.com/freedomofpress/securedrop/pull/6506

## Description of Changes

- Standard patch-level release updates
- Standard kernel troubleshooting guide
- Tails 5 reminder

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [x] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

**Comment by gonzalo-bulnes** [comment]
I assume that we'll tag the `main` branch once this PR is merged, so I haven't added any tag.

File: docs/upgrade/2.4.1_to_2.4.2.rst
Comment (eloquence) [inline docs/upgrade/2.4.1_to_2.4.2.rst:99]: We can bump the version to 2.4.2 here as well.

**Comment by eloquence** [comment]
> I assume that we'll tag the main branch once this PR is merged, so I haven't added any tag.

Yep that's correct, we will tag the `main` branch with `2.4.2` once the release is live and these docs changes have landed. This will trigger a new build of the stable version of the documentation.

File: docs/upgrade/2.4.1_to_2.4.2.rst
Comment (gonzalo-bulnes) [reply docs/upgrade/2.4.1_to_2.4.2.rst:99]: Yes! Thank you @eloquence I completely missed this one.

Comment (gonzalo-bulnes) [reply docs/upgrade/2.4.1_to_2.4.2.rst]: Fixed :white_check_mark:

**Comment by gonzalo-bulnes** [comment]
(Fixed the remaining `2.4.1` mention and rebased.)

**Review by eloquence — APPROVED** [review]
Looks great, thanks! Good to merge once release is live. :)

**Review by legoktm — APPROVED** [review]

**Pull request merged**
